aclocal.m4

```diff
@@ -45,10 +45,20 @@ AC_DEFUN([OCAML_CC_VENDOR], [
   AC_MSG_CHECKING([C compiler vendor])
   AC_PREPROC_IFELSE(
     [AC_LANG_SOURCE([
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && defined(__clang_major__) && defined(__clang_minor__)
+msvc _MSC_VER clang __clang_major__ __clang_minor__
```

Comment:
Looking at the mingw block below, it seems it is acceptable for this macro to return its output on multiple lines, so why not use the same logic for msvc? I.e.
```
#if defined(_MSC_VER)
msvc _MSC_VER
#if defined(clang stuff)
clang suffix
#endif
#endif
```